code/obj/item/mops_cleaners.dm

```diff
@@ -208,14 +208,17 @@ WET FLOOR SIGN
 
 	if(src.reagents.has_reagent("water") || src.reagents.has_reagent("cleaner"))
 		JOB_XP(user, "Janitor", 2)
-
 	var/obj/decal/D = new/obj/decal(get_turf(src))
 	D.name = "chemicals"
 	D.icon = 'icons/obj/chemical.dmi'
 	D.icon_state = "chempuff"
 	D.create_reagents(5) // cogwerks: lowered from 10 to 5
 	src.reagents.trans_to(D, 5)
 	playsound(src.loc, "sound/effects/zzzt.ogg", 50, 1, -6)
+	// Direct cleaning for objects or items directly sprayed in containers are not cleaned
+	if (istype(A, /obj/item))
+		D.reagents.reaction(A)
+		D.reagents.remove_any(1)
```

Comment:
Might want to rework the proc so that the ranged spray doesn't happen if spraying something directly

Reply:
Possibly check if the item is in contents? It seems like the overall intention for the spray is to clean multiple tiles and items in one go.

Reply:
Reworked this to do a turf check.